test/observer.test.js

```diff
@@ -54,50 +41,68 @@ describe("nestedRendering", async () => {
             }
         }
     )
-    beforeAll(async done => {
-        // the side-effect in  does not views alive when using static rendering test suite
-        useStaticRendering(false)
-        await asyncReactDOMRender(<TodoList />, testRoot)
-        done()
+
+    beforeEach(() => {
```

Comment:
Personally, I find it safer instead of using `beforeEach` to simply call a function in each test and return those variables from that. That way I don't need to rely on closure variables. In my opinion, any shared state makes tests bit more flaky.

Reply:
I agree with you, but I intentionally tried to minimize the surface of the changes. We will be able to refactor these tests in the future, especially since I have a desire to improve something else in them.